src/main/java/com/google/api/gax/core/PropertiesProvider.java

```diff
@@ -56,14 +55,33 @@ public static String getGrpcVersion() {
     return grpcVersion != null ? grpcVersion : DEFAULT_VERSION;
   }
 
+  /**
+   * Utility method of retrieving the property value of the given key from a property file in the
+   * package.
+   *
+   * @param loadedClass The class used to get the resource path
+   * @param propertyFilePath The relative file path to the property file
+   * @param key Key string of the property
+   */
+  public static String loadProperty(Class<?> loadedClass, String propertyFilePath, String key) {
+    try {
+      Properties properties = new Properties();
+      properties.load(loadedClass.getResourceAsStream(propertyFilePath));
```

Comment:
That seems fine to me... I don't know that we want to cache this at such a generic level.